src/graph.rs

```diff
@@ -1164,22 +1156,58 @@ mod tests {
         graph.insert(Edge::new(4, 5));
         graph.insert(Edge::new(5, 3));
 
-        let between_map = graph.betweenness_centrality(1);
+        let between_map = graph.betweenness_centrality(1, false);
         let close_map = graph.closeness_centrality(1);
-        let mut betweenness: [f64; 6] = [0.0; 6];
-        let mut closeness: [f64; 6] = [0.0; 6];
-        for i in 0..6 {
+        const N: usize = 6;
+        let mut betweenness = [0.0; N];
+        let mut closeness = [0.0; N];
+        for i in 0..N {
             betweenness[i] = *between_map.get(&i).unwrap();
             closeness[i] = *close_map.get(&i).unwrap();
         }
 
         let total_path_length = [9, 8, 8, 7, 7, 9];
-        let mut expected_closeness: [f64; 6] = [0.0; 6];
+        let mut expected_closeness = [0.0; N];
         let expected_betweenness: [f64; 6] = [1.0, 1.5, 1.5, 2.5, 2.5, 0.0];
-        for i in 0..6 {
-            expected_closeness[i] = total_path_length[i] as f64 / 5.0;
+        for i in 0..N {
+            expected_closeness[i] = total_path_length[i] as f64 / (N - 1) as f64;
+        }
+
+        assert_eq!(betweenness, expected_betweenness);
+        assert_eq!(closeness, expected_closeness);
+    }
+
+    #[test]
+    fn bhatia_graph_normalized() {
+        let mut graph: Graph<usize> = Graph::new();
+        // this graph reproduces the image at:
+        // https://www.youtube.com/watch?v=ptqt2zr9ZRE
+        graph.insert(Edge::new(0, 1));
+        graph.insert(Edge::new(1, 3));
+        graph.insert(Edge::new(3, 4));
+        graph.insert(Edge::new(4, 2));
+        graph.insert(Edge::new(2, 0));
+        graph.insert(Edge::new(4, 5));
+        graph.insert(Edge::new(5, 3));
+
+        let between_map = graph.betweenness_centrality(1, true);
+        let close_map = graph.closeness_centrality(1);
+        const N: usize = 6;
+        let mut betweenness = [0.0; N];
+        let mut closeness = [0.0; N];
+        for i in 0..N {
+            betweenness[i] = *between_map.get(&i).unwrap();
+            closeness[i] = *close_map.get(&i).unwrap();
         }
 
+        let total_path_length = [9, 8, 8, 7, 7, 9];
+        let mut expected_closeness = [0.0; N];
+        let mut expected_betweenness = [1.0, 1.5, 1.5, 2.5, 2.5, 0.0];
+        const DIVISOR: f64 = ((N - 1) * (N - 2) / 2) as f64;
```

Comment:
It looks like divisor in 126 line in betweenness.rs file but here is additional division by 2 - line 126 does not have such division - why?

Reply:
Good to question that!  However, the code it correct:
1. The short answer is, the `/2` is already in the data (`expected_betweenness`).  For example, we have our `randomish_graph` with 7 nodes.  So in the `betweenness.rs` code, we divide the results by 2, or by 30 (6x5).  In the test data though, the `expected_betweenness` contains the non-normalized results, i.e., **already divided-by-two**.  So, we only need to divide that by 15, not 30.  I.e.:  `(N-1)(N-2)/2`
2. I confirmed that the results match what the python library is sending when the normalized flag is set
![Screenshot 2023-03-27 at 09 28 02](https://user-images.githubusercontent.com/15172321/227872733-14669e06-7136-4715-8afc-445af0f398c3.png)

This is what out test code outputs:
```
compute_closeness: done 62.958µs
  normalized: i 0 = 0.03333333333333333
  normalized: i 1 = 0.6333333333333333
  normalized: i 2 = 0.6
  normalized: i 3 = 0.13333333333333333
  normalized: i 4 = 0
  normalized: i 5 = 0.13333333333333333
  normalized: i 6 = 0
ok
```
I hope this is clear.

Reply:
I just checked in a change, so the second test case with normalization uses the DIVISOR same as in the first case.  😄

Reply:
Yes, it's clear :)